src/docdb/nodes.ts

```diff
@@ -0,0 +1,186 @@
+	get iconPath(): any {
+		return {
+			light: path.join(__filename, '..', '..', '..', '..', 'resources', 'icons', 'light', 'Database.svg'),
+			dark: path.join(__filename, '..', '..', '..', '..', 'resources', 'icons', 'dark', 'Database.svg')
+		};
+	}
+
+	readonly collapsibleState = vscode.TreeItemCollapsibleState.Collapsed;
+
+	getDbLink(): string {
+		return 'dbs/' + this.id;
+	}
+
+	getChildren(): Promise<INode[]> {
+		return this.getCollections();
+	}
+
+	async getCollections(): Promise<INode[]> {
+		let dbLink: string = this.getDbLink();
+		let collections;
+		let parentNode = this;
+		let client = new DocumentDBConnectionClient(this.server.getEndpoint(), {masterKey: this.server.getPrimaryMasterKey() });
+		try {
+			collections = await this.listCollections(dbLink, client);
+		} catch (err) {
+			collections = [];
+			vscode.window.showErrorMessage(err.code + ": " + JSON.parse(err.body).message);
+		}
+
+		return collections.map(collection => new DocDBCollectionNode(collection.id, parentNode));
+	}
+
+	async listCollections(databaseLink, client): Promise<any> {
+		let collections = await client.readCollections(databaseLink);
+		return await new Promise<any[]>((resolve, reject) => {
+			collections.toArray( (err, cols: Array<Object>) => err ? reject(err) : resolve(cols) );
+		});
+	}
+
+}
+
+
+export class DocDBCollectionNode implements INode {
+
+	constructor(readonly id: string, readonly db: DocDBDatabaseNode) {
+	}
+
+	readonly contextValue: string = 'DocDbCollection';
+	readonly DocGenerationJSONReplacer = function replacer(key, value) {
+		let redundantProperties: Array<string> = ["_rid", "_ts", "_self", "_etag", "_attachments"]; 
+		if(redundantProperties.indexOf(key) > -1){
+			return undefined;
+		}
+		return value;
+	}
+
+	get label(): string {
+		return this.id;
+	}
+
+	get iconPath(): any {
+		return {
+			light: path.join(__filename, '..', '..', '..', '..', 'resources', 'icons', 'light', 'Collection.svg'),
+			dark: path.join(__filename, '..', '..', '..', '..', 'resources', 'icons', 'dark', 'Collection.svg'),
+		};
+	}
+	readonly collapsibleState = vscode.TreeItemCollapsibleState.None;
+
+	readonly command: Command = {
+		command: 'cosmosDB.openDocDBCollection',
+		arguments: [this],
+		title: ''
+	};
+
+
+	async getDocuments(): Promise<any> {
+		let dbLink: string = this.db.getDbLink();
+		let client = new DocumentDBConnectionClient(this.db.server.getEndpoint(), {masterKey: this.db.server.getPrimaryMasterKey() });
+		let docs;
+		try{
+			let collSelfLink = dbLink + "/colls/" + this.id;
+			let docs = await this.readOneCollection(collSelfLink, client);
+			console.log(docs);
+			return await docs; 
+		} catch (error){
+			console.log(error.message);
+			return;
+		}
+	}
+
+	async readOneCollection(selfLink, client): Promise<any>{
+		let documents = await client.readDocuments(selfLink);
+		return await new Promise<any[]>((resolve, reject) => {
+			documents.toArray((err, docs: Array<Object>) => err ? reject(err) : resolve(docs));
```

Comment:
As discussed in standup, we need to limit the max number of documents. Maybe we should use the 'executeNext' function instead of 'toArray' (Also when you call `readDocuments`, you can pass in `FeedOptions` that has a `maxItemCount`)